app/assets/javascripts/school_administrator_dashboard/absences_dashboard/SchoolwideAttendance.js

```diff
@@ -42,30 +67,36 @@ class SchoolwideAttendance extends React.Component {
   //There's no application awareness of valid school days, but there are almost never schoolwide perfect attendance days
   //We use days the school has recorded at least one absence as a proxy for school days
   schoolDays() {
-    return Object.keys(this.props.schoolAbsenceEvents);
+    return Object.keys(this.props.schoolAbsenceEventsByDay);
   }
 
   schoolYearDateRange() {
     //change this to change the maximum date range available for the dashboard
-    const fullYearDateRange = Object.keys(this.schoolAverageDailyAttendance()).sort();
+    const fullYearDateRange = Object.keys(this.props.schoolAbsenceEventsByDay).sort();
     const today = moment.utc();
     return DashboardHelpers.filterDates(fullYearDateRange, DashboardHelpers.schoolYearStart(), today);
   }
 
   render() {
+    const studentsByHomeroom = DashboardHelpers.groupByHomeroom(this.props.dashboardStudents);
+    const schoolUnexcusedAbsenceEventsByDay = this.schoolUnexcusedAbsenceEventsByDay();
     return (
         <SchoolAbsenceDashboard
           schoolAverageDailyAttendance = {this.schoolAverageDailyAttendance()}
-          homeroomAverageDailyAttendance = {this.homeroomAverageDailyAttendance()}
-          schoolAbsenceEvents = {this.props.schoolAbsenceEvents}
+          schoolAverageDailyAttendanceUnexcused = {this.schoolAverageDailyAttendanceUnexcused(schoolUnexcusedAbsenceEventsByDay)}
+          homeroomAverageDailyAttendance = {this.homeroomAverageDailyAttendance(this.homeroomAbsenceEventsByDay(studentsByHomeroom))}
+          homeroomAverageDailyAttendanceUnexcused = {this.homeroomAverageDailyAttendance(this.HomeroomAbsenceEventsByDayunexcused(studentsByHomeroom))}
+          schoolAbsenceEventsByDay = {this.props.schoolAbsenceEventsByDay}
+          schoolUnexcusedAbsenceEventsByDay = {schoolUnexcusedAbsenceEventsByDay}
```

Comment:
This looks to me like it is calculating everything for both conditions all the time.  I'm wondering if there are ways to simplify this do calculate only what is needed based on the user's selection.  Are there ways we could push this filtering upward, so that the component doesn't change - we apply a filter to the absence events, and the rest of the code stays unchanged?

Looking more closely at the component, I'm wondering if maybe having the separation between placing the computation in `SchoolwideAttendance` and the state about the filters in `SchoolAbsenceDashboard` is what's driving this.  Concretely, would it help simplify if `SchoolAbsenceDashboard` were the main component, that owned the bits of state, and then it just called into the various functions it needed to compute things (that now live in `SchoolwideAttendance`?  It seems like that might remove the need to compute both things.

Or if the idea that computing both is intended as a performance optimization related to the comment [here](https://github.com/studentinsights/studentinsights/pull/1811/files#diff-6f3e04125c9062d8f9711609485b94a8R8) about responsiveness?

Reply:
@kevinrobinson yeah, you have this right. The trouble here is that we have to go from students -> absences -> daily averages -> monthly averages. In early versions of the dashboard, those were computed with each state change and it was irritatingly slow. Not unusable, but not ideal. If the dashboard gets the daily averages as a prop, this is much faster. However, if we start to do more with filtering based on other absence attributes, this model gets unwieldily. 

I agree this is kind of clumsy. One thought might be to allow filtering in the class list (which seems like the most useful part of the component) and not in the charts, which is the piece that cares about these averages. I hadn't come up with a good way to communicate that visually, though, so I haven't tried it. 

One important point is that this problem is unique to the absence dashboard and tardies/discipline have a more streamlined functional approach.

Reply:
@edavidsonsawyer Thanks, this is super helpful!  and now I'm more on the same page with how this works and what you're thinking 👍

We could make these improvements to simplify, but let's keep moving forward.  I'll look into the data quality on this tomorrow morning and see what comes up there.  Assuming that's mostly in good shape, then we can 🚢 this tomorrow, and we can see what other needs come up down the line (eg, next time we add something, if the factoring makes that change more involved, maybe do some refactoring then).